standard-libraries/tests/list-tests.3.mr

```diff
@@ -0,0 +1,86 @@
+## Tests for generic lists
+
+test list-append-test()
+  var List{String} list
+  list.append(owner "first")
+  list.append(owner "second")
+  list.append(owner "third")
+  var ListIterator{String} iter
+  iter.init(user list)
```

Comment:
Right now you can still use a user-defined iterator, just have the list initialize it.